starknet_py/tests/e2e/contract_interaction/interaction_test.py

```diff
@@ -7,18 +7,73 @@
 from starknet_py.contract import Contract
 from starknet_py.net.client import BadRequest
 from starknet_py.net.models import InvokeFunction
-from starknet_py.tests.e2e.utils import DevnetClient
+from starknet_py.tests.e2e.utils import DevnetClient, DevnetClientNoAccount
 from starknet_py.utils.crypto.facade import sign_calldata
 
 directory = os.path.dirname(__file__)
 
 map_source = Path(directory, "map.cairo").read_text("utf-8")
 
 
+@pytest.mark.asyncio
+async def test_auto_fee_estimation():
+    client = await DevnetClient.make_devnet_client()
+    key = 2
+    value = 3
+
+    deployment_result = await Contract.deploy(
+        client=client, compilation_source=map_source
+    )
+    deployment_result = await deployment_result.wait_for_acceptance()
+    contract = deployment_result.deployed_contract
+    contract = await Contract.from_address(contract.address, client)
+    invocation = await contract.functions["put"].invoke(key, value, auto_estimate=True)
+    await invocation.wait_for_acceptance()
+    (response,) = await contract.functions["get"].call(key)
+
+    assert response == value
+
+
+@pytest.mark.asyncio
+async def test_max_fee_takes_precedence_over_auto_setimate():
+    client = await DevnetClient.make_devnet_client()
+    key = 2
+    value = 3
+
+    deployment_result = await Contract.deploy(
+        client=client, compilation_source=map_source
+    )
+    deployment_result = await deployment_result.wait_for_acceptance()
+    contract = deployment_result.deployed_contract
+    contract = await Contract.from_address(contract.address, client)
+
+    invocation = contract.functions["put"].prepare(
+        key, value, max_fee=0, auto_estimate=True
```

Comment:
I think we should throw when both are provided.

Reply:
This actually is a bug in test, prepare no longer accepts `auto_estimate`. Though i can throw in invoke if both are provided